From 52b55877cafb899e138500a7cdd8e91e00793656 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@stgraber.org>
Date: Tue, 28 Apr 2026 22:39:20 +0200
Subject: [PATCH] tpm_library: Remove const qualifier on overwriten start
 variable
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@stgraber.org>
---
 src/tpm_library.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tpm_library.c b/src/tpm_library.c
index 7b2ea687..f48f4fd3 100644
--- a/src/tpm_library.c
+++ b/src/tpm_library.c
@@ -435,7 +435,7 @@ static unsigned char *TPMLIB_GetPlaintext(const char *stream,
                                           const char *endtag,
                                           size_t *length)
 {
-    const char *start, *end;
+    char *start, *end;
     unsigned char *plaintext = NULL;
 
     start = strstr(stream, starttag);
-- 
2.47.3

