diff --git a/src/lxc/start.c b/src/lxc/start.c
index 8daa7d73c..f24812db7 100644
--- a/src/lxc/start.c
+++ b/src/lxc/start.c
@@ -1420,6 +1420,11 @@ static int do_start(void *data)
 	if (!lxc_switch_uid_gid(new_uid, new_gid))
 		goto out_warn_father;
 
+	ret = prctl(PR_SET_DUMPABLE, prctl_arg(1), prctl_arg(0),
+			prctl_arg(0), prctl_arg(0));
+	if (ret < 0)
+		goto out_warn_father;
+
 	ret = lxc_ambient_caps_down();
 	if (ret < 0) {
 		ERROR("Failed to clear ambient capabilities");
