diff --git a/configure.ac b/configure.ac
index 42cd198..54d62b9 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,6 +1,6 @@
 dnl Process this file with autoconf 2.69 or later to produce
 dnl a configure script.
-AC_PREREQ([2.71])
+AC_PREREQ([2.69])
 AC_INIT
 AC_CONFIG_SRCDIR([config/config.h.in])
 AC_CONFIG_HEADERS([config/config.h])
@@ -175,8 +175,6 @@ AC_C_INLINE
 AC_C_RESTRICT
 
 dnl Checks for header files.
-AC_CHECK_INCLUDES_DEFAULT
-
 PA_ADD_HEADERS(string.h)
 PA_ADD_HEADERS(stdarg.h)
 AC_CHECK_HEADERS(inttypes.h)
