From f1d6301d794727d39cd3e2504cc56c0063b6c8c0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@stgraber.org>
Date: Wed, 22 Nov 2023 23:16:13 +0000
Subject: [PATCH 6/6] [incus] Remove Canonical image servers
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@stgraber.org>
---
 src/pages/images/ImageSelector.tsx | 12 ------------
 1 file changed, 12 deletions(-)

diff --git a/src/pages/images/ImageSelector.tsx b/src/pages/images/ImageSelector.tsx
index 2d85c7e..e4e5ded 100644
--- a/src/pages/images/ImageSelector.tsx
+++ b/src/pages/images/ImageSelector.tsx
@@ -81,16 +81,6 @@ const ImageSelector: FC<Props> = ({ onSelect, onClose }) => {
     },
   );
 
-  const { data: canonicalImages = [], isLoading: isCiLoading } = useQuery({
-    queryKey: [queryKeys.images, canonicalServer],
-    queryFn: () => loadImages(canonicalJson, canonicalServer),
-  });
-
-  const { data: minimalImages = [], isLoading: isMinimalLoading } = useQuery({
-    queryKey: [queryKeys.images, minimalServer],
-    queryFn: () => loadImages(minimalJson, minimalServer),
-  });
-
   const { data: localImages = [], isLoading: isLocalImageLoading } = useQuery({
     queryKey: [queryKeys.images, project],
     queryFn: () => fetchImageList(project ?? ""),
@@ -109,8 +99,6 @@ const ImageSelector: FC<Props> = ({ onSelect, onClose }) => {
     ? []
     : localImages
         .map(localLxdToRemoteImage)
-        .concat([...minimalImages].reverse().sort(byLtsFirst))
-        .concat([...canonicalImages].reverse().sort(byLtsFirst))
         .concat(linuxContainerImages)
         .filter((image) => archSupported.includes(image.arch));
 
-- 
2.34.1

