From b429729297ed2bf93af12b5f429f5c0122e61a02 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?St=C3=A9phane=20Graber?= <stgraber@stgraber.org>
Date: Wed, 22 Nov 2023 23:16:13 +0000
Subject: [PATCH 5/5] Remove Canonical image servers
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Stéphane Graber <stgraber@stgraber.org>
---
 src/pages/images/ImageSelector.tsx | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/src/pages/images/ImageSelector.tsx b/src/pages/images/ImageSelector.tsx
index 3af5f69..868a7fe 100644
--- a/src/pages/images/ImageSelector.tsx
+++ b/src/pages/images/ImageSelector.tsx
@@ -81,25 +81,13 @@ const ImageSelector: FC<Props> = ({ onSelect, onClose }) => {
     },
   );
 
-  const { data: canonicalImages = [], isLoading: isCiLoading } = useQuery({
-    queryKey: [queryKeys.images, canonicalServer],
-    queryFn: () => loadImages(canonicalJson, canonicalServer),
-  });
-
-  const { data: minimalImages = [], isLoading: isMinimalLoading } = useQuery({
-    queryKey: [queryKeys.images, minimalServer],
-    queryFn: () => loadImages(minimalJson, minimalServer),
-  });
-
   const { data: localImages = [], isLoading: isLocalImageLoading } = useQuery({
     queryKey: [queryKeys.images, project],
     queryFn: () => fetchImageList(project ?? ""),
   });
 
   const isLoading =
-    isCiLoading ||
     isLciLoading ||
-    isMinimalLoading ||
     isLocalImageLoading ||
     isSettingsLoading;
   const archSupported = getArchitectureAliases(
@@ -110,8 +98,6 @@ const ImageSelector: FC<Props> = ({ onSelect, onClose }) => {
     : localImages
         .filter((image) => !image.cached)
         .map(localLxdToRemoteImage)
-        .concat([...minimalImages].reverse().sort(byLtsFirst))
-        .concat([...canonicalImages].reverse().sort(byLtsFirst))
         .concat(linuxContainerImages)
         .filter((image) => archSupported.includes(image.arch));
 
-- 
2.39.2

